### DISCUSSION TITLE
Slug - Title - Filename synchronization and updates

### DISCUSSION THREAD

**Discussion by Dekker500** [estruyf/vscode-front-matter#579] [General]
I'll start by describing a scenario, then get to my current problem...

frontmatter.json:
```
  "frontMatter.taxonomy.alignFilename": true,
  "frontMatter.taxonomy.slugPrefix": "my-prefix",
  "frontMatter.taxonomy.contentTypes": [
    {
      "name": "default",
      "filePrefix": "yyww",
      "pageBundle": true,
      "previewPath": null,
      "template": "[[workspace]]/archetypes/article.md",
      "fields": [
        {
          "title": "Title",
          "name": "title",
          "type": "string"
        },
        {
          "title": "Slug",
          "name": "slug",
          "type": "slug",
          "editable": true,
          "default": "{{slug}}"
        },
...
```

i. From FM dashboard, click "Create new content"
ii. Choose/enter "where you want to create your content" (say, Posts)
iii. Give it a Title (say, "Fancy New Article")
iv. FM auto-generates a new file, pre-populated with the generated slug [fancy-article], and with Filename [2317-fancy-new-article.md]

This is where the problems start...

1. Why doesn't the filename use the slug that was automatically generated? `"frontMatter.taxonomy.alignFilename": true` was set. [problem]
2. If I click "Optimize slug" under actions, it switches to the auto-generated slug, and all appears good [no problem]
3. If I MANUALLY change the Slug in the frontmatter/metadata, the side panel takes few seconds to update, but finally it shows the new slug. Problem is the FILENAME is never updated! [problem]
4. If I click the "update available" icon next to the slug box in the side panel, it seems to re-generate the optimized slug. This is either a bug, or poorly tool-tipped feature. It should read "optimize slug" then. Instead, I would have expected it to update the filename based on the "current" slug text... [problem or documentation]

Any insights into the 3 related questions I have? (note num 2 is not really a question but is an observation)

**Comment by estruyf** [comment]
Related to the following issue: https://github.com/estruyf/vscode-front-matter/issues/519
